app/v1/controllers/requestParsers/SubmitReturnRequestParser.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers
+
+import javax.inject.Inject
+import uk.gov.hmrc.domain.Vrn
+import v1.controllers.requestParsers.validators.SubmitReturnValidator
+import v1.models.request.submit.{SubmitRawData, SubmitRequest, SubmitRequestBody}
+
+class SubmitReturnRequestParser @Inject()(val validator: SubmitReturnValidator)
+  extends RequestParser[SubmitRawData, SubmitRequest] {
+
+  override protected def requestFor(data: SubmitRawData): SubmitRequest = {
+    SubmitRequest(Vrn(data.vrn), data.body.asJson.get.as[SubmitRequestBody])
```

Comment:
The reads/writes for the SubmitRequestBody case class are not correct. "totalVatDue" should map to "vatDueTotal", and "netVatDue" should map to "vatDueNet". Also the "agentReferenceNumber", and "receivedAt" fields which are required for the downstream request to DES are missing.

Reply:
isn't agentreference being added when auth is implemented?

Reply:
It still needs to be in the model.